src/Storages/IStorage.h

```diff
@@ -302,6 +312,9 @@ class IStorage : public std::enable_shared_from_this<IStorage>, public TypePromo
     /// Description of virtual columns. Optional, may be set in constructor.
     MultiVersionVirtualsDescriptionPtr virtuals;
 
+    /// Description of all virtual columns, include common virtual columns.
+    MultiVersionVirtualsDescriptionPtr all_virtuals;
```

Comment:
Maybe we can avoid adding a new set somehow...
Can we check whether `getVirtualsXXX` contains `_table` column, and process all the logic at the planner level?

Reply:
@KochetovNicolai Deleted all_virtual, and implemented by maintaining a global common virtual column in IStorage.